mobile/lib/main.dart

```diff
@@ -272,13 +272,26 @@ class _TenTenOneAppState extends State<TenTenOneApp> {
       await walletChangeNotifier.refreshWalletInfo();
     } on FfiException catch (error) {
       FLog.error(text: "Failed to initialise: Error: ${error.message}", exception: error);
+      exitAppAfterErrors();
     } catch (error) {
       FLog.error(text: "Failed to initialise: $error", exception: error);
+      exitAppAfterErrors();
     } finally {
       FlutterNativeSplash.remove();
     }
   }
 
+  exitAppAfterErrors() {
+    FLog.info(text: "Exiting app due to startup errors");
+
+    // Add arbitrary amount of sleep to allow accepting permission dialogs if
+    // lack of permissions caused the app to crash, so that the user can start the app next time
+    var duration = const Duration(seconds: 10);
+    sleep(duration);
+
+    exit(1); // Hopefully this triggers TestFlight crash report
```

Comment:
I guess it will crash the app, but I guess we will only get a crash report in TestFlight with the StackTrace pointing to this exit. Why not remove the try catch from above?

Reply:
> > but I'm not sure whether they would pop up over the splashscreen
> 
> For me it did pop up over the splash screen.

great, thanks for testing.

> 
> > Ideally I would have guarded it with a feature flag (managed as per feature flag / A/B testing discussion), especially given our TestFlight release cycle.
> 
> I don't think that this fix justifies a feature flag. If we would use feature flags for little things like that I'd fear we would quickly end up with loads of flags, which I guess will get hell to maintain. I think the best course of action here is to write a simple test, that fails the intitialisation and assert that the app crashes.

the feature flag idea is for *temporary* adding things that are potentially risky, 
Such feature flag can be removed in the upcoming release after it stabilises.

Sometimes a test would do, it's not a panacea though.

> 
> > the alternative approach could involve storing a flag in preferences initialised = false at startup, and setting to true at the end of init, and then adjusting welcome screen based on that.
> 
> I think we are fine with the splash screen only being removed once the app is initialised, if we ensure the app crashes in case the initialisation failed.

Reply:
closing this conversation, as removing the try/catch does not close the app, and the app just hangs.